### PR TITLE
Update BoringSSL to 7c522995d1ea5386b3223a19b0f62e73c1f76b17

### PR DESCRIPTION
**Pull request by Lukasa**
Also remove a temporary BoringSSL patch and deal with no C files being
in fiat/

### CODE REVIEW

File: scripts/vendor-boringssl.sh [hunk 1]
```diff
@@ -191,7 +191,7 @@ PATTERNS=(
 'crypto/*/*/*.S'
 'crypto/*/*/*/*.c'
 'third_party/fiat/*.h'
-'third_party/fiat/*.c'
+#'third_party/fiat/*.c'
```
Comment (glbrntt) [inline scripts/vendor-boringssl.sh:194]: If this remains commented out and files matching this pattern _do_ exist the next time this is run, will there be an obvious failure?

Comment (Lukasa) [reply scripts/vendor-boringssl.sh:194]: Yeah, the compile will fail due to missing symbols.

Comment (glbrntt) [reply scripts/vendor-boringssl.sh:194]: Cool, thanks

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Pull request merged**